include/re/re.h

```diff
@@ -147,6 +147,46 @@ re_comp(enum re_dialect dialect,
 	const struct fsm_options *opt,
 	enum re_flags flags, struct re_err *err);
 
+enum re_strings_flags {
+	RE_STRINGS_ANCHOR_LEFT  = 1 << 0,
+	RE_STRINGS_ANCHOR_RIGHT = 1 << 1,
+
+	/* This signals that the automaton should be a standard Aho-Corasick
+	 * automaton.  An Aho-Corasick automaton graph 
+	 * is equivalent to RE_STIRNGS_ANCHOR_RIGHT, but
+	 * the execution is slightly different.
+	 *
+	 *
+	 * With standard FSM automatons, a match is only reported when the FSM
+	 * ends on an end state.
+	 *
+	 * In Aho-Corasick, each time an end state is encountered, the state
+	 * machine should record/report a match.  This is not yet implemented
+	 * and intended for future development.
+	 */
+	RE_STRINGS_AC_AUTOMATON = 1 << 2
+};
+
+struct fsm *
+re_strings(const struct fsm_options *opt, const char *sv[], size_t n, enum re_strings_flags flags);
+
+struct re_strings_builder;
```

Comment:
After merge, I might expose some opaque handle to the trie to libre's API, and get rid of this struct.